packages/router/src/utils/index.ts

```diff
@@ -37,3 +37,6 @@ export const noop = () => {}
  */
 export const isArray: (arg: ArrayLike<any> | any) => arg is ReadonlyArray<any> =
   Array.isArray
+
+export const isObject = (val: unknown): val is Record<any, any> =>
+  val !== null && typeof val === 'object'
```

Comment:
Usually I write this one
```suggestion
  val && typeof val === 'object'
```

I'm curious if you prefer this version for any particular reason

Reply:
I'd usually write it the way you suggested.

I wasn't sure about the TS types for that function so I just copied the equivalent function in Vue core:

- https://github.com/vuejs/core/blob/f66a75ea75c8aece065b61e2126b4c5b2338aa6e/packages/shared/src/general.ts#L52

I'm happy either way.

Reply:
I guess you could just use isRouteLocation where isObject is used in router.ts